File: java/src/main/java/com/languageweaver/sdk/samples/translations/text/EdgeTextTranslationService.java
package com.languageweaver.sdk.samples.translations.text;

import com.languageweaver.sdk.common.constants.Format;
import com.languageweaver.sdk.common.edge.EdgeLanguageWeaverClient;
import com.languageweaver.sdk.translate.edge.request.EdgeTranslateTextRequest;
import com.languageweaver.sdk.translate.edge.result.EdgeTranslationTextResult;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class EdgeTextTranslationService {

    public static void main(String[] args) throws Exception {
        try (EdgeLanguageWeaverClient lwClient = new EdgeLanguageWeaverClient().build()) {
            translateText(lwClient);
            translateTextWithLinguisticOptions(lwClient);
        }
    }

    private static void translateText(EdgeLanguageWeaverClient lwClient) throws IOException, InterruptedException {
        EdgeTranslateTextRequest translateTextRequest = new EdgeTranslateTextRequest()
                .setLanguagePairId("EngFra_Generic_SRV_TNMV_8_5_x_1")
                .addInput("The weather is wonderful today!")
                .setInputFormat(Format.PLAIN)
                .addDictionary("DictionaryName1")
                .addDictionary("DictionaryName2");
        final EdgeTranslationTextResult translateTextResult = lwClient.translateText(translateTextRequest);

        System.out.println(translateTextResult.getTranslation());
    }

    private static void translateTextWithLinguisticOptions(EdgeLanguageWeaverClient lwClient) throws IOException, InterruptedException {
        Map<String, String> linguisticOptions = new HashMap<>();
        linguisticOptions.put("formality", "Informal");
        EdgeTranslateTextRequest translateTextRequest = new EdgeTranslateTextRequest()
                .setLanguagePairId("EngJpn_Generic_SRV_TNMV_8_5_x_6")
                .addInput("It was a long journey, so I'm really tired now")
                .setLinguisticOptions(linguisticOptions)
                .setInputFormat(Format.PLAIN)
                .addDictionary("DictionaryName1")
                .addDictionary("DictionaryName2");
        final EdgeTranslationTextResult translateTextResult = lwClient.translateText(translateTextRequest);

        System.out.println(translateTextResult.getTranslation());
    }
}

File: dotnet/Samples/Translations/Text/EdgeTextTranslationService.cs
using System.Text;
using LanguageWeaver.Sdk.Common.Edge;
using LanguageWeaver.Sdk.Constants;
using LanguageWeaver.Sdk.Translate.Edge.Request;
using LanguageWeaver.Sdk.Translate.Edge.Result;

namespace LanguageWeaver.Sdk.Samples.Translations.Text;

public static class EdgeTextTranslationService
{
    public static void Main()
    {
        using var lwClient = new EdgeLanguageWeaverClient().Build();

        TranslateText(lwClient);
        TranslateTextWithLinguisticOptions(lwClient);
    }

    private static void TranslateText(EdgeLanguageWeaverClient lwClient)
    {
        var translateTextRequest = new EdgeTranslateTextRequest
        {
            LanguagePairId = "EngFra_Generic_SRV_TNMV_8_5_x_1",
            Input = new List<string>
            {
                "The weather is wonderful today!"
            },
            InputFormat = Format.Plain,
            Dictionaries = new List<string>
            {
                "DictionaryName1",
                "DictionaryName2"
            }
        };

        EdgeTranslationTextResult translateTextResult = lwClient.TranslateText(translateTextRequest);
        Console.WriteLine(translateTextResult.Translation);
    }

    private static void TranslateTextWithLinguisticOptions(EdgeLanguageWeaverClient lwClient)
    {
        Dictionary<string, string> linguisticOptions = new Dictionary<string, string>();
        linguisticOptions.Add("formality", "Formal");

        var translateTextRequest = new EdgeTranslateTextRequest()
        {
            LanguagePairId = "EngJpn_Generic_SRV_TNMV_8_5_x_6",
            Input = new List<string> { "It was a long journey, so I'm really tired now" },
            LinguisticOptions = linguisticOptions,
            InputFormat = Format.Plain,
            Dictionaries = new List<string>
            {
                "DictionaryName1",
                "DictionaryName2"
            }
        };

        EdgeTranslationTextResult translateTextResult = lwClient.TranslateText(translateTextRequest);
        Console.OutputEncoding = Encoding.UTF8;
        Console.WriteLine(translateTextResult.Translation);
    }
}